api/app/routers/topics.py

```diff
@@ -53,42 +55,164 @@ def get_topics(
         }
         return hashlib.md5(json.dumps(data, sort_keys=True).encode()).hexdigest()
     """
-    return search_topics(db, current_user.user_id)
-
-
-# TODO: This endpoint should be merged into get_topics after implementing pagination
-@router.get("/search", response_model=List[schemas.TopicEntry])
-def search_topics_(
-    title_words: Optional[List[str]] = Query(None),
-    abstract_words: Optional[List[str]] = Query(None),
-    threat_impacts: Optional[List[int]] = Query(None),
-    misp_tag_ids: Optional[List[UUID]] = Query(None),
-    tag_ids: Optional[List[UUID]] = Query(None),
+    return get_topics_internal(db, current_user.user_id)
+
+
+@router.get("/search", response_model=schemas.SearchTopicsResponse)
+def search_topics(
+    offset: int = Query(0, ge=0),
+    limit: int = Query(10, ge=1, le=100),  # 10 is default in web/src/pages/TopicManagement.jsx
+    sort_key: schemas.TopicSortKey = Query(schemas.TopicSortKey.THREAT_IMPACT),
+    threat_impacts: str = Query(""),
+    topic_ids: str = Query(""),
+    title_words: str = Query(""),
+    abstract_words: str = Query(""),
+    tag_names: str = Query(""),
+    misp_tag_names: str = Query(""),
+    zone_names: str = Query(""),
+    creator_ids: str = Query(""),
```

Comment:
default=None にするのではダメでしょうか？ (あと max_lengthも追加)

手元で試した感じでは、引数なしはNone、引数ありで空文字列は空文字列と使い分けられそうです
以下は引数ありで空文字列の場合
```
"GET /topics/search?offset=0&limit=10&sort_key=threat_impact&tag_names= HTTP/1.1" 200 OK
```

```suggestion
    threat_impacts: Optional[str] = Query(default=None, max_length=50),
    topic_ids: Optional[str] = Query(default=None, max_length=50),
    title_words: Optional[str] = Query(default=None, max_length=50),
    abstract_words: Optional[str] = Query(default=None), max_length=50,
    tag_names: Optional[str] = Query(default=None, max_length=50),
    misp_tag_names: Optional[str] = Query(default=None, max_length=50),
    zone_names: Optional[str] = Query(default=None, max_length=50),
    creator_ids: Optional[str] = Query(default=None, max_length=50),
```